From f5727e1b7c8e9ccf57de625820534566f1e14a76 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Wed, 29 Nov 2017 11:08:14 +0200
Subject: [PATCH 2201/2241] ppv2: dts: a8kp: increase number of PPv2 interrupts
 to 9

CP110 can support up to 9 PPv2 interrupts.
In A7K and A8K platforms with up to 4 CPU only 5 interrupts were used.
Single queue mode:
4 - TX Done interrupt for each CPU
1 - Shared RX interrupt for all CPU's.
Multi queue mode:
4 - TX Done&RX interrupt for each CPU

A810 has 8/16/32 CPU's, to enhance resource utilization all 9 interrupts
would be used in A810 platform.

Change-Id: Ie78bee6472e61aa9dcdf17e8de4423f3c06d3c6e
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/47026
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi      | 43 ++++++++++++++--------
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |  2 +-
 2 files changed, 28 insertions(+), 17 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 9961d92..59b0a4d 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -473,32 +473,43 @@
 				clock-names = "pp_clk", "gop_core_clk", "gop_clk", "mg_core_clk", "mg_clk";
 				status = "okay";
 				CP110_LABEL(eth0): eth0@010000 {
-					interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-							<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-							<ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-							<ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-							<ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					interrupts = <ICU_GRP_NSR 39 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 43 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 47 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 51 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 55 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 59 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 63 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 67 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 71 IRQ_TYPE_LEVEL_HIGH>;
 					port-id = <0>; /* pp2_port_id */
 					emac-data = <&CP110_LABEL(emac0)>;
 					status = "disabled";
 				};
 				CP110_LABEL(eth1): eth1@020000 {
-					interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-							<ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-							<ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-							<ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-							<ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
+					interrupts = <ICU_GRP_NSR 40 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 44 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 48 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 52 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 56 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 60 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 64 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 68 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 72 IRQ_TYPE_LEVEL_HIGH>;
 					port-id = <1>; /* pp2_port_id */
 					emac-data = <&CP110_LABEL(emac2)>;
 					status = "disabled";
 				};
 				CP110_LABEL(eth2): eth2@030000 {
-					interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#0 Interrupt */
-							<ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#1 Interrupt */
-							<ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#2 Interrupt */
-							<ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>, /* TX CPU#3 Interrupt */
-							<ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>; /* RX Shared Interrupt */
-
+					interrupts = <ICU_GRP_NSR 41 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 45 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 49 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 53 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 57 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 61 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 65 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 69 IRQ_TYPE_LEVEL_HIGH>,
+							<ICU_GRP_NSR 73 IRQ_TYPE_LEVEL_HIGH>;
 					port-id = <2>; /* pp2_port_id */
 					emac-data = <&CP110_LABEL(emac3)>;
 					status = "disabled";
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 9a5661b..5428fce 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5709,7 +5709,7 @@ static struct mv_pp2x_platform_data pp22_pdata = {
 	.multi_hw_instance = true,
 	.mv_pp2x_port_queue_vectors_init = mv_pp22_queue_vectors_init,
 	.mv_pp2x_port_isr_rx_group_cfg = mv_pp22_port_isr_rx_group_cfg,
-	.num_port_irq = 5,
+	.num_port_irq = 9,
 	.hw.desc_queue_addr_shift = MVPP22_DESC_ADDR_SHIFT,
 };
 
-- 
2.7.4

